arcade/tests/tool/test_create_tool_definition_errors.py

```diff
@@ -41,6 +41,14 @@ def func_with_multiple_context_params(context: ToolContext, context2: ToolContex
     pass
 
 
+@tool(
+    desc="A function with a required secret with a missing key (illegal)",
+    requires_secrets=[""],
+)
+def func_with_missing_secret_key(context: ToolContext):
+    pass
+
+
```

Comment:
Within `Catalog.create_tool_definition`, we should ensure `isinstance(secret.key_id, str)`

Suggesting adding a test for this:
```py
@tool(
    desc="A function that requires a secret",
    requires_secrets=[True],
)
def func_with_invalid_type_secret_requirement():
    pass
```

Reply:
Great catch, thanks. Added test